apps/emqx/src/emqx_hookpoints.erl

```diff
@@ -89,6 +108,12 @@
 %%
 %% By default, callbacks are executed in the channel process context.
 
+-callback 'alarm.activated'(alarm_activated_context(), rule_engine_conf()) ->
```

Comment:
That's not quite clear. It seems that alarms and hooks are in the `emqx` app and out of the context of the rule engine 🤔

Reply:
The rule engine just wants to receive alarm info, that's why it's a new hook.

Reply:
I mean, emqx app doesn't even know that any rule engine exist.

Reply:
The callback signatures just document the arguments in absence of the additional hook arguments, so we probably may just remove the `rule_engine_conf()` arg from the specs.

Reply:
Ah, I see now what you mean.

I've removed mentions to that config, even though that particular callback impl injects it in the hook.